### PR TITLE
fix: clean up mediastream plugins

### PR DESCRIPTION
**Pull request by KaustubhKumar05**
- [x] Test on mweb and desktop

### CODE REVIEW

File: packages/hms-video-store/src/plugins/video/HMSMediaStreamPluginsManager.ts [hunk 1]
```diff
@@ -63,4 +63,8 @@ export class HMSMediaStreamPluginsManager {
   getPlugins(): string[] {
     return Array.from(this.plugins).map(plugin => plugin.getName());
   }
+
+  async cleanup() {
+    this.plugins.forEach(plugin => plugin.stop());
```
Comment (raviteja83) [inline packages/hms-video-store/src/plugins/video/HMSMediaStreamPluginsManager.ts:68]: we could call removePlugins no?

File: packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts [hunk 2]
```diff
@@ -264,6 +264,7 @@ export class HMSLocalVideoTrack extends HMSVideoTrack {
     super.cleanup();
     this.transceiver = undefined;
     await this.pluginsManager.cleanup();
+    await this.mediaStreamPluginsManager.cleanup();
```
Comment (raviteja83) [inline packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts:267]: shouldn't we call this before super.cleanup?

Comment (KaustubhKumar05) [reply packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts:267]: Yeah. The videop lugins manager cleanup is fine at L266 or should I move that up as well?

File: packages/hms-video-store/src/plugins/video/HMSMediaStreamPluginsManager.ts (hunk 1 above)
Comment (KaustubhKumar05) [reply packages/hms-video-store/src/plugins/video/HMSMediaStreamPluginsManager.ts]: Yeah, updated

**Review by raviteja83 — DISMISSED** [review]

File: packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts [hunk 3]
```diff
@@ -261,9 +261,10 @@ export class HMSLocalVideoTrack extends HMSVideoTrack {
    */
   async cleanup() {
     this.removeTrackEventListeners(this.nativeTrack);
+    await this.mediaStreamPluginsManager.cleanup();
```
Comment (amar-1995) [inline packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts:264]: Why we called it before `super.cleanup()`? If there is any specific reason please add a comment.

Comment (KaustubhKumar05) [reply packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts:264]: Cleaning the track before stopping the plugin could break third party plugins. Adding a comment

**Review by ygit — DISMISSED** [review]

**Review by amar-1995 — APPROVED** [review]

**Review by raviteja83 — APPROVED** [review]

**Pull request merged**